clash-prelude/src/Clash/Intel/ClockGen.hs

```diff
@@ -49,78 +68,255 @@ import Clash.Promoted.Symbol (SSymbol)
 import Clash.Signal.Internal
   (Signal, Clock, Reset, KnownDomain, HasAsynchronousReset)
 
+{- $domains
+Synthesis domains are denoted by the type-parameter
+@dom :: t'Clash.Signal.Domain'@ as occuring in for instance
+@t'Clash.Signal.Signal' dom a@; see "Clash.Signal" for more information. For
+each domain, there is only a single clock signal which clocks that domain;
+mixing clock signals is a design error. Conversely, it is possible to clock
+multiple domains using the same clock signal, in complex designs.
+
+For the clock generator inputs, create a domain with the correct clock frequency
+and reset polarity. For instance, if the clock input is a free-running clock at
+a frequency of 50 MHz (a period of 20 ns or 20,000 ps), and the reset input
+connected to the clock generator is /active-low/, the following will instantiate
+the required input domain:
+
+@
+'Clash.Signal.createDomain' 'Clash.Signal.vSystem'{vName=\"DomInput\", vPeriod=20000, vResetPolarity='Clash.Signal.ActiveLow'}
+@
+
+If you haven't determined the frequency you want the design to run at, the
+predefined 100 MHz domain t'Clash.Signal.XilinxSystem' can be a good starting
+point. The datasheet for your FPGA specifies lower and upper limits, but the
+true maximum frequency is determined by your design.
+
+Supposing you need a clock running at 150 MHz for your design, the following
+will instantiate a suitable domain:
+
+@
+'Clash.Signal.createDomain' 'Clash.Signal.vXilinxSystem'{vName=\"Dom150\", vPeriod='Clash.Signal.hzToPeriod' 150e6}
+@
+
+@Dom150@ will have 'Clash.Signal.Synchronous' resets on its memory elements
```

Comment:
Perhaps leave this part out, only from Stratix 10 onwards does Intel recommend synchronous resets.